src/validation.cpp

```diff
@@ -1810,6 +1810,8 @@ PackageMempoolAcceptResult MemPoolAccept::AcceptPackage(const Package& package,
         AcceptSubPackage(txns_package_eval, args);
     PackageValidationState& package_state_final = multi_submission_result.m_state;
 
+    ClearSubPackageState();
```

Comment:
This shouldn't be necessary, but shouldn't hurt either. `AcceptSubPackage` calls it already for each subpackage.

Reply:
I think the issue is if we `quit_early`; I recall the fuzzer hitting the new `Assume()` calls via `LimitMempoolSize()` at the line just below, before I added this.

Reply:
Was the assert failed because of the changeset state, or something that could be hit in master?

`AcceptSubPackage` is called for each submission attempt of any subpackage, so I'm not sure why `quit_early` would be problematic.

Reply:
still curious about this if you figure out how it was hit

Reply:
Now I think I may have been mistaken; I've commented out this line and started fuzzing again, and I cannot reproduce the crash. Also, I agree with your assessment after re-reading the code some more.

However, it would be nice if the SubPackageState were clearly destroyed when it goes out of scope from logic that uses it...  Even though it's seemingly not necessary to call cleanup here, it makes me a little uncomfortable to leave the code in a place where it takes some investigation to figure out that the changeset will be gone by the time LimitMempoolSize() is called.

I might just add a comment mentioning that this should have gotten cleaned up by `AcceptSubPackage()` but that I'm leaving the cleanup call for readability, given that the call to `LimitMempoolSize` requires that the changeset be gone.  Does that seem reasonable?